string obj_name

string frame